distributed/scheduler.py

```diff
@@ -4907,7 +4917,31 @@ async def update_graph(
                 start=start,
                 stimulus_id=stimulus_id or f"update-graph-{start}",
             )
-        except RuntimeError as e:
+            task_state_created = time()
+            metrics.update(
+                {
+                    "start": start,
+                    "duration_materialization": materialization_done - start,
+                    "duration_ordering": materialization_done - ordering_done,
+                    "duration_state_initialization": ordering_done - task_state_created,
+                    "duration_total": task_state_created - start,
+                }
+            )
+            evt_msg = {
+                "action": "update-graph",
+                "stimulus_id": stimulus_id,
+                "metrics": metrics,
+                "status": "OK",
+            }
+            self.log_event(["all", client, "update-graph"], evt_msg)
+            logger.debug("Task state created. %i new tasks", len(self.tasks) - before)
+        except Exception as e:
+            evt_msg = {
+                "action": "update-graph",
+                "stimulus_id": stimulus_id,
+                "status": "error",
+            }
```

Comment:
This is an unrelated change but it shouldn't be too disruptive for the review

Reply:
I don't think this is very useful and I don't want to add the exception to this. This is primarily supposed to be a stream of metrics and I don't like adding Exception objects or large strings to it. I also find the logging and proper exception handling mechanism should be sufficient.

If this is a contentious topic I will remove this change from the PR